.github/workflows/ci-mlir.yml

```diff
@@ -25,6 +25,12 @@ jobs:
     - uses: actions/checkout@v3
       with:
         path: xdsl
+        
+    - name: Install native dependencies
+      uses: awalsh128/cache-apt-pkgs-action@latest
+      with:
+        packages: mesa-vulkan-drivers
+        version: 1.0
```

Comment:
Personally, I would prefer to separate our testing environments a bit. Otherwise, we might run into versioning issues were certain packages are required, but not built with just the default config.

For that reason, I think I would have prefered this to be its own separate CI run.

What do others think? I guess our requirements-optional.txt and what is needed where is a big mess already, so this might not just be an issue with these tests, but something deeper. tagging @math-fehr @superlopuh and @PapyChacal here :)

Reply:
I would love to look into structuring those things for sure, but probably later I. August as we all have deadlines running currently 🙂

Reply:
I would be interested in splitting the requirements-optional per subproject, with `extras` installing them all.